scala_proto/private/proto_to_scala_src.bzl

```diff
@@ -0,0 +1,48 @@
+load(
+    "//scala/private:common.bzl",
+    "write_manifest_file",
+)
+load("//scala/private:rule_impls.bzl", "compile_scala")
+load("//scala_proto/private:dep_sets.bzl", "SCALAPB_DEPS", "GRPC_DEPS")
+
+
+def _root_path(f):
+    if f.is_source:
+        return f.owner.workspace_root
+    return "/".join([f.root.path, f.owner.workspace_root])
+
+
+def _colon_paths(data):
+    return ":".join([
+        f.path
```

Comment:
I've just handled this differently, the external locations all work right in the tests i've done. Instead of a big chain of imports in the rule i just move things to where they are expected. For me it removed a bunch of warnings about `external/com_google_..` not existing. Seemed like it made it simpler though i'm not wed to it.